lib/Dialect/FIRRTL/Transforms/GrandCentralTaps.cpp

```diff
@@ -271,6 +271,288 @@ static FailureOr<Literal> parseIntegerLiteral(MLIRContext *context,
   return lit;
 }
 
+/// Find the lowest-common-ancestor `lcaModule`, between `srcTarget` and
+/// `wireTarget`, and set `pathFromSrcToWire` with the path between them through
+/// the `lcaModule`.
+static LogicalResult
+findLCAandSetPath(AnnoPathValue &srcTarget, AnnoPathValue &wireTarget,
+                  SmallVector<InstanceOp> &pathFromSrcToWire,
+                  FModuleOp &lcaModule, ApplyState &state) {
+
+  auto srcModule = cast<FModuleOp>(srcTarget.ref.getModule());
+  auto wireModule = cast<FModuleOp>(wireTarget.ref.getModule());
+  // Get the path from top to `srcModule` and `wireModule`, and then compare the
+  // path to find the lca. Then use that to get the path from `srcModule` to the
+  // `wireModule`.
+  auto *top = state.instancePathCache.instanceGraph.getTopLevelNode();
```

Comment:
Nit: This isn't necessary if the `if` block below exits. This can be moved after the `if` block.